src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -441,7 +441,9 @@ internal class DefaultRazorSemanticTokensInfoService : RazorSemanticTokensInfoSe
             Models.RazorSemanticTokensEdit[] edits)
         {
             var updatedTokens = previousCSharpTokens.ToList();
-            foreach (var edit in edits)
+
+            // C# returns edits that are expected to be applied in reverse order.
+            foreach (var edit in edits.Reverse())
```

Comment:
Is there a way we can cheaply detect this in case additional people start returning edits to us?

Reply:
Good idea! I added a debug assert in my latest push.